impeller/renderer/backend/vulkan/context_vk.cc

```diff
@@ -95,7 +95,16 @@ std::shared_ptr<ContextVK> ContextVK::Create(Settings settings) {
   return context;
 }
 
-ContextVK::ContextVK() = default;
+namespace {
+thread_local uint64_t tls_context_count = 0;
+uint64_t CalculateHash(void* ptr) {
+  // You could make a context once per nanosecond for 584 years on one thread
+  // before this overflows.
+  return ++tls_context_count;
+}
+}  // namespace
+
+ContextVK::ContextVK() : hash_(CalculateHash(this)) {}
```

Comment:
Would it be simpler to just use the address of the object instead?

Reply:
(really I think we want a weak_ptr instead)

Reply:
> Would it be simpler to just use the address of the object instead?

That's what it was before and that doesn't work because eventually you'll get a duplicate address and everything will come crashing down (details in the linked issue).

> (really I think we want a weak_ptr instead)

It can't be a weak_ptr because the way the vulkan backend was designed is that this global variable is supposed to keep the pool alive until the thread is destroyed.  One of the first things I did was try to make it be a weak_ptr but that would lead to a crash on the worker thread.  I'm still seeing racy crashes so this may have to be changed eventually.  I'm just trying to make the original design not crash.